cmd/re/main.go

```diff
@@ -0,0 +1,259 @@
+	envAuthTimeout     = "MF_AUTH_GRPC_TIMEOUT"
+	envClientTLS       = "MF_RE_CLIENT_TLS"
+	envCACerts         = "MF_RE_CA_CERTS"
+	envThingsLocation  = "MF_RE_THINGS_LOCATION"
+	envMfBSURL         = "MF_RE_BS_SVC_URL"
+	envMfCertsURL      = "MF_RE_CERTS_SVC_URL"
+	envTLS             = "MF_RE_ENV_CLIENTS_TLS"
+)
+
+type config struct {
+	logLevel        string
+	httpPort        string
+	kuiperURL       string
+	authHTTPPort    string
+	authGRPCPort    string
+	jaegerURL       string
+	serverCert      string
+	serverKey       string
+	singleUserEmail string
+	singleUserToken string
+	authURL         string
+	authTimeout     time.Duration
+	clientTLS       bool
+	caCerts         string
+
+	ThingsLocation string
+	MfBSURL        string
+	MfCertsURL     string
+	TLS            bool
+}
+
+func main() {
+	cfg := loadConfig()
+
+	logger, err := logger.New(os.Stdout, cfg.logLevel)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+
+	authTracer, authCloser := initJaeger("auth", cfg.jaegerURL, logger)
+	defer authCloser.Close()
+	auth, _ := createAuthClient(cfg, authTracer, logger)
+
+	tracer, reCloser := initJaeger("re", cfg.jaegerURL, logger)
+	defer reCloser.Close()
+
+	SDKCfg := mfSDK.Config{
+		BaseURL:           cfg.ThingsLocation,
+		BootstrapURL:      cfg.MfBSURL,
+		CertsURL:          cfg.MfCertsURL,
+		HTTPAdapterPrefix: "http",
+		MsgContentType:    "application/json",
+		TLSVerification:   cfg.TLS,
+	}
+	SDK := mfSDK.NewSDK(SDKCfg)
+
+	svc := newService(cfg.kuiperURL, auth, SDK, logger)
+	errs := make(chan error, 2)
+
+	go startHTTPServer(rehttpapi.MakeHandler(tracer, svc), cfg.httpPort, cfg, logger, errs)
+
+	go func() {
+		c := make(chan os.Signal)
+		signal.Notify(c, syscall.SIGINT)
+		errs <- fmt.Errorf("%s", <-c)
+	}()
+
+	err = <-errs
+	logger.Error(fmt.Sprintf("Re service terminated: %s", err))
+}
+
+func loadConfig() config {
+	tls, err := strconv.ParseBool(mainflux.Env(envClientTLS, defClientTLS))
+	if err != nil {
+		log.Fatalf("Invalid value passed for %s\n", envClientTLS)
+	}
+
+	authTimeout, err := time.ParseDuration(mainflux.Env(envAuthTimeout, defAuthTimeout))
+	if err != nil {
+		log.Fatalf("Invalid %s value: %s", envAuthTimeout, err.Error())
+	}
+
+	return config{
+		logLevel:        mainflux.Env(envLogLevel, defLogLevel),
+		httpPort:        mainflux.Env(envHTTPPort, defHTTPPort),
+		kuiperURL:       mainflux.Env(envKuiperURL, defKuiperURL),
+		serverCert:      mainflux.Env(envServerCert, defServerCert),
+		serverKey:       mainflux.Env(envServerKey, defServerKey),
+		jaegerURL:       mainflux.Env(envJaegerURL, defJaegerURL),
+		singleUserEmail: mainflux.Env(envSingleUserEmail, defSingleUserEmail),
+		singleUserToken: mainflux.Env(envSingleUserToken, defSingleUserToken),
+		authURL:         mainflux.Env(envAuthURL, defAuthURL),
+		authTimeout:     authTimeout,
+		clientTLS:       tls,
+		caCerts:         mainflux.Env(envCACerts, defCACerts),
+		MfBSURL:         mainflux.Env(envMfBSURL, defMfBSURL),
```

Comment:
do you need these properties for certs and bootstrap?

Reply:
I need this for the sdk. Is there a simpler way?

Reply:
@darkodraskovic not sure, please do the research. I think it should be possible if you do not use TLS in the SDK, which is probably not needed now, as all services are in the private network

Reply:
In any case do you need new ENVARS like `MF_RE_BS_SVC_URL` ?